src/main/java/io/jenkins/plugins/appcenter/api/AppCenterServiceFactory.java

```diff
@@ -88,6 +89,56 @@ public UploadService createUploadService(@Nonnull final String uploadUrl) {
         return retrofit.create(UploadService.class);
     }
 
+    public BlobClient createBlobUploadService(@Nonnull final String uploadUrl) {
```

Comment:
I opted to configure the two Http clients in the same place - the AppCenterServiceFactory. This is because we can configure the proxy settings in one place. In the future I also want to move Azure BlobClient to use OkHttp as its Http engine because then we can reuse even more of the configuration we already have. For now I think that might kick up a lot of unrelated changes because I need to update the libraries first. Hence these workarounds.